oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/DurationUtils.java

```diff
@@ -213,4 +213,36 @@ private DateTime parseToDateTime(Downsampling downsampling, long time) {
         }
         throw new UnexpectedException("Unexpected downsampling: " + downsampling.name());
     }
+
+    public DateTime startTimeBucket2DateTime(Downsampling downsampling, long startTB) {
```

Comment:
First, time covert should be inside `TimeBucket`, take a look at InfluxDB, https://github.com/apache/skywalking/pull/4239/files#diff-df3b7b3f55b25fb0e29e345e7e6cc078. Maybe good for you.
I think you don't need the `DateTime`? Directly convert to long as day/month makes more sense?

Reply:
Currently, you covert to DateTime, then cover to long like `202002`, but actually, you could use divide to get `202002` directly and more quickly, right?

Reply:
Indeed, it maybe faster to convert to long. When the date while loop statement is added, it needs to be converted to the character format of the date.

Reply:
Why do you need to add? The current prefix should be enough. `2020021015`-`2020021315` to days is 20200210`-`20200213`

Do I miss anything?

Reply:
When crossing months, you need to consider the big and small months. In addition, you need to consider some leap years. like `20200227-20200305`

Reply:
OK, got it, make sense.